janitor/timeseries.py

```diff
@@ -176,3 +180,273 @@ def sort_timestamps_monotonically(
 
     # Return the dataframe
     return df
+
+
+def _flag_jumps_single_col(
+    df: pd.DataFrame,
+    col: str,
+    scale: str,
+    direction: str,
+    threshold: Union[int, float],
```

Comment:
@ericmjl 
I've seen that using `Union[int, float]` is not recommended as the numeric tower treats `int` as `float`. Thus, saying `float` here is equivalent to saying `Union[int, float]`. Do you have strong feelings on that? Happy to update for best practices.

[PEP 484](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower)
>**The numeric tower**
PEP 3141 defines Python's numeric tower, and the stdlib module numbers implements the corresponding ABCs (Number, Complex, Real, Rational and Integral). There are some issues with these ABCs, but the built-in concrete numeric classes complex, float and int are ubiquitous (especially the latter two :-).
>Rather than requiring that users write import numbers and then use numbers.Float etc., this PEP proposes a straightforward shortcut that is almost as effective: when an argument is annotated as having type float, an argument of type int is acceptable; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable. This does not handle classes implementing the corresponding ABCs or the fractions.Fraction class, but we believe those use cases are exceedingly rare.

Reply:
@loganthomas interesting, thanks for bringing up the "numeric tower", as it's the first time I've heard of it! The PEP3141 being marked as status "Final" - does this mean it's been accepted?

Reply:
@ericmjl 
I thought the term "numeric tower" was interesting as well... This actually came up on my Twitter feed this week ([here](https://twitter.com/raymondh/status/1306387490048098304?s=20)).

As far as a status of "Final", I assume this snippet from [PEP](https://www.python.org/dev/peps/pep-0001/#:~:text=Once%20a%20PEP%20has%20been,be%20changed%20to%20%22Final%22.) 1 still holds true? 🤷‍♂️ 
> Once a PEP has been accepted, the reference implementation must be completed. When the reference implementation is complete and incorporated into the main source code repository, the status will be changed to "Final".

Reply:
> As far as a status of "Final", I assume this snippet from PEP 1 still holds true? 🤷‍♂️

Probably so!

> Do you have strong feelings on that? Happy to update for best practices.

@loganthomas I was wondering how commonplace knowledge of the numeric tower is amongst developers and users -- if we change all `Union[int, float]`s to `float`, would we be not adhering to "explicit is better than implicit"? My hunch says that yes, we would be breaking the "explicit is better than implicit" guidance in the Zen of Python. Leaving the type annotation as `Union[int, float]` might help users a bit more, perhaps?